FirebaseAppCheck/Sources/Public/FirebaseAppCheck/FIRAppCheckAvailability.h

```diff
@@ -46,29 +46,9 @@
 
 #pragma mark - App Attest
 
-// App Attest availability was extended to macOS and Mac Catalyst in Xcode 12.5.
-#if (defined(__IPHONE_14_5) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_5) || \
-    (defined(__MAC_11_3) && __MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_11_3) ||        \
-    (defined(__TVOS_14_5) && __TV_OS_VERSION_MAX_ALLOWED >= __TVOS_14_5)
-
-// Targets where `DCAppAttestService` is available to be used in preprocessor conditions.
-#define FIR_APP_ATTEST_SUPPORTED_TARGETS TARGET_OS_IOS || TARGET_OS_OSX
-
-// `AppAttestProvider` availability annotations
-#define FIR_APP_ATTEST_PROVIDER_AVAILABILITY \
-  API_AVAILABLE(macos(11.0), ios(14.0)) API_UNAVAILABLE(tvos, watchos)
-
-#else  // (defined(__IPHONE_14_5) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_5) || \
-          (defined(__MAC_11_3) && __MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_11_3) ||        \
-          (defined(__TVOS_14_5) && __TV_OS_VERSION_MAX_ALLOWED >= __TVOS_14_5)
-
 // Targets where `DCAppAttestService` is available to be used in preprocessor conditions.
-#define FIR_APP_ATTEST_SUPPORTED_TARGETS TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+#define FIR_APP_ATTEST_SUPPORTED_TARGETS TARGET_OS_IOS || TARGET_OS_OSX || TARGET_OS_TV
 
 // `AppAttestProvider` availability annotations
 #define FIR_APP_ATTEST_PROVIDER_AVAILABILITY \
-  API_AVAILABLE(ios(14.0)) API_UNAVAILABLE(macos, tvos, watchos)
-
-#endif  // (defined(__IPHONE_14_5) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_5) || \
-          (defined(__MAC_11_3) && __MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_11_3) ||        \
-          (defined(__TVOS_14_5) && __TV_OS_VERSION_MAX_ALLOWED >= __TVOS_14_5)
+  API_AVAILABLE(macos(11.0), ios(14.0), tvos(15.0)) API_UNAVAILABLE(watchos)
```

Comment:
non-blocking: This is similar to the DeviceCheck situation where App Attest isn't actually functional on macOS 11.0. Apple's documentation says macOS 11.0+ on the [DCAppAttestService](https://developer.apple.com/documentation/devicecheck/dcappattestservice) page but all of the methods/properties list 11.3+ (e.g., [`shared`](https://developer.apple.com/documentation/devicecheck/dcappattestservice/3573914-shared), [`isSupported`](https://developer.apple.com/documentation/devicecheck/dcappattestservice/3573915-issupported), and [`generateKey(completionHandler:)`](https://developer.apple.com/documentation/devicecheck/dcappattestservice/3573913-generatekey)). The availability values are also different for Mac Catalyst than iOS as well ([`14.5+`](https://developer.apple.com/documentation/devicecheck/dcappattestservice/3573915-issupported)).

We may want to audit these as part of [FR#10095](https://github.com/firebase/firebase-ios-sdk/issues/10095) for Firebase 10.

Reply:
Added a comment in #10095